src/map/status.c

```diff
@@ -1560,6 +1560,41 @@ static int status_fixed_revive(struct block_list *bl, unsigned int per_hp, unsig
 	return 1;
 }
 
+/**
+ * check 'disabled_skills' flag settings from `db\(pre-)re\map_zone_db.conf` use in 'status_check_skilluse' function which allow for plugin manipulation
+ * @param src        The source object can be BL_PC, BL_MOB, BL_PET, BL_HOM, BL_MER, BL_ELEM type
+ * @param st         The status_data of the source object, use to check MD_BOSS of BL_MOB type
+ * @param skill_id   The skill ID use
+ * @return 1 on success, 0 otherwise
+ **/
+static bool status_check_skilluse_mapzone(struct block_list *src, struct status_data *st, uint16 skill_id)
+{
+	nullpo_ret(src);
+	nullpo_ret(st);
+	for (int i = 0; i < map->list[src->m].zone->disabled_skills_count; i++) {
+		if (skill_id == map->list[src->m].zone->disabled_skills[i]->nameid && (map->list[src->m].zone->disabled_skills[i]->type&src->type)) {
+			if (src->type == BL_PC) {
+				struct map_session_data *sd = BL_CAST(BL_PC, src);
```

Comment:
because here was already check for ``BL_PC``, better use ``BL_UCAST(BL_PC, src)``. ``BL_UCAST`` not do type check